damnit/api.py

```
@@ -171,6 +173,110 @@ def summary(self):
+                if isinstance(obj, h5py.Group):
+                    xarray_group = obj.name
+                else:
+                    dset = obj
+            elif data_fallback:
+                # Implicit: use data as preview if suitable
+                grp = f[self.name]
+                type_hint = self._type_hint(grp)
+                if type_hint is DataType.DataArray:
+                    xarray_group = self.name
+                elif type_hint is DataType.Dataset:
+                    return None
+                else:
+                    dset = grp['data']
+            else:
+                return None
+
+            if xarray_group is not None:
+                for obj in f[xarray_group].values():
+                    if isinstance(obj, h5py.Dataset) and (
+                        obj.ndim > 3 or (obj.ndim == 3 and obj.shape[-1] not in (3, 4))
+                    ):
+                        return None  # Too many dims: bail out before loading
+
+                import xarray as xr
+                arr = xr.load_dataarray(
+                    self._h5_path, group=xarray_group, engine="h5netcdf"
+                )
+                if arr.ndim != 0 and np.issubdtype(arr.dtype, np.number):
+                    return arr
+
+            elif np.issubdtype(dset.dtype, np.number) and (
+                    dset.ndim in (1, 2) or (dset.ndim == 3 and dset.shape[-1] in (3, 4))
+            ):
+                value = dset[()]
+
+                if type_hint is DataType.PlotlyFigure:
+                    import plotly.io as pio
+                    b = value.tobytes()
+                    return pio.from_json(b) if deserialize_plotly else b.decode()
+                else:
+                    return value
+
+        return None
+
+    def preview(self):
+        """Show the preview data
+
+        This is intended for use in Jupyter notebooks
+        """
+        if (obj := self.preview_data()) is None:
+            return
```

Could we return a string like "Preview could not be generated"? That would be a bit clearer than the function seeming to do nothing when called.

---

I agree it makes sense to have something visible in that case; what do you think about throwing an error rather than returning a string?

---

Yep that seems fine.